src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1195,16 +1195,155 @@ private ImplicitNamedTypeSymbol BindScriptClass()
             return namespaceOrType as ImplicitNamedTypeSymbol;
         }
 
-        internal Imports GlobalImports
+        /// <summary>
+        /// Usings that are not from source.  For regular compilations, these are the global usings.
+        /// For submissions, global and declared usings from previous submissions are also included.
+        /// </summary>
+        /// <remarks>
+        /// Always consider whether usings should be considered before accessing this (e.g. not when binding other usings).
+        /// </remarks>
+        internal ImmutableArray<NamespaceOrTypeAndUsingDirective> ExternalUsings => _externalUsings.Value;
+
+        private ImmutableArray<NamespaceOrTypeAndUsingDirective> BindExternalUsings()
         {
-            get { return _globalImports.Value; }
+            var globalImports = Imports.FromGlobalUsings(this);
+            globalImports.Complete(CancellationToken.None);
+
+            var newUsings = globalImports.Usings;
+
+            var previousSubmission = this.PreviousSubmission;
+            if (previousSubmission == null)
+            {
+                return newUsings;
+            }
+
+            var externalUsings = ArrayBuilder<NamespaceOrTypeAndUsingDirective>.GetInstance();
+            var uniqueTargets = PooledHashSet<NamespaceOrTypeSymbol>.GetInstance();
+
+            foreach (var previousUsing in previousSubmission.ExternalUsings)
+            {
+                var previousTarget = previousUsing.NamespaceOrType;
+                if (previousTarget.IsType)
+                {
+                    if (uniqueTargets.Add(previousTarget))
```

Comment:
This loop does not consider using aliases.  They're handled separately during lookup.